### PR TITLE
build: upgrade @mui/x-data-grid to v6

### PR DESCRIPTION
**Pull request by sbolel**
## Summary

This PR upgrades the `@mui/x-data-grid` package from v5 to v6, a major version upgrade that introduces some backward incompatible changes. This upgrade is preparation for Upload UI v2, enabling the use of the newest features of MUI Data Grid. To accommodate these changes, the `disableSelectionOnClick` prop, no longer supported in v6, has been removed from all instances where it was used. Additionally, the `GridColumns` import has been renamed to `GridColDef`, following the changes in the package's API.

Additionally, an explicit type has been added to the `useLoaderData` call in the `TeamView` component.

### Added

- Added explicit type to `useLoaderData` call in `TeamView` component.

### Changed

- Upgraded `@mui/x-data-grid` to v6.
- Renamed deprecated `GridColumns` import to `GridColDef` across the codebase.

### Deprecated

- _n/a_

### Removed

- Removed the `disableSelectionOnClick` prop from all instances of the MUI `Grid` component where it was used.

### Fixed

- _n/a_

## How to test

Given that the changes involve an upgrade to a major package version and modifications to the codebase to accommodate the upgrade, a thorough test of all the components that use the `Grid` component would normally be necessary. However, since none of the components are actually being rendered after the deprecation of the v1 UI, there is nothing that needs manual verification.

1. Pull the changes from this branch.
2. Run `yarn install` to update the project dependencies.
3. Run `yarn test` to confirm tests still pass.
4. Run `yarn build` to confirm build still succeeds.

### CODE REVIEW

**Review by DerekStrickland — APPROVED** [review]
LGTM

**Comment by sbolel** [comment]
Merging without waiting for second approval after rebasing `main` into this branch dismissed the approval -- the only conflicts that were resolved were in `yarn.lock`, which happened automatically as a result of running `yarn install`.

**Pull request merged**
